rosa/cli.py

```diff
@@ -140,98 +140,33 @@ def get_available_flags(command):
-                        )
-                        commands_dict[top_command][command][_command][
-                            "region"
-                        ] = check_flag_in_flags(
-                            command_list=[rosa_cmd, top_command, _command],
-                            flag_str=region_str,
-                        )
-                else:
-                    # If sub command doesn't have sub command
-                    commands_dict[top_command][command][
-                        "json_output"
-                    ] = check_flag_in_flags(
-                        command_list=[rosa_cmd, top_command, command],
-                        flag_str=output_flag_str,
-                    )
-                    commands_dict[top_command][command][
-                        "auto_answer_yes"
-                    ] = check_flag_in_flags(
-                        command_list=[rosa_cmd, top_command, command],
-                        flag_str=auto_answer_yes_str,
-                    )
-                    commands_dict[top_command][command][
-                        "auto_mode"
-                    ] = check_flag_in_flags(
-                        command_list=[rosa_cmd, top_command, command],
-                        flag_str=auto_mode_str,
-                    )
-                    commands_dict[top_command][command]["region"] = check_flag_in_flags(
-                        command_list=[rosa_cmd, top_command, command],
-                        flag_str=region_str,
-                    )
-
-        else:
-            # If top command doesn't have sub command
-            commands_dict[top_command]["json_output"] = check_flag_in_flags(
-                command_list=[rosa_cmd, top_command],
-                flag_str=output_flag_str,
+def parse_help():
+    commands_dict = benedict()
+
+    def _fill_commands_dict_with_support_flags(flag_key_path):
+        support_commands = {
+            "json_output": "-o, --output",
+            "auto_answer_yes": "-y, --yes",
+            "auto_mode": "-m, --mode",
+            "region": "--region",
+        }
+        for cli_flag, flag_value in support_commands.items():
+            commands_dict[flag_key_path][cli_flag] = check_flag_in_flags(
+                command_list=["rosa"] + flag_key_path,
+                flag_str=flag_value,
             )
-            commands_dict[top_command]["auto_answer_yes"] = check_flag_in_flags(
-                command_list=[rosa_cmd, top_command],
-                flag_str=auto_answer_yes_str,
-            )
-            commands_dict[top_command]["auto_mode"] = check_flag_in_flags(
-                command_list=[rosa_cmd, top_command],
-                flag_str=auto_mode_str,
-            )
-            commands_dict[top_command]["region"] = check_flag_in_flags(
-                command_list=[rosa_cmd, top_command],
-                flag_str=region_str,
+
+    def _build_command_tree(commands_search_path):
```

Comment:
`_build_command_tree` and `_fill_commands_dict_with_support_flags` should get `commands_dict` arg